### PR TITLE
AT:7491 - Ensuring all resources are attached to a VPC as necessary

### PR DESCRIPTION
**Pull request by jeffsegal-ccpo**
This change leverages the existing `network` parameter and pushes it down to be available to use when creating `NodeJsFunction` instances such that they include a `vpc` and `vpcSubnets` property if the given environment includes a Network stack (and thus `network` is defined). 

Ticket: AT-7491

### CODE REVIEW

**Comment by kylelaker-ccpo** [comment]
This seems reasonable. Did you happen to test by modifying the pipeline stack to pass down the `network` prop?

**Review by kylelaker-ccpo — CHANGES_REQUESTED** [review]
So the more I think about this, the more I want to make sure that we're generally following a pattern of not passing "bigger" construct types around.

So literals/primitives/resolvables, constructs, and stacks can get passed to stacks and apps but Constructs should only ever have literals/primitives/resolvables passed to them (so let's not pass a stack to a construct).

File: lib/constructs/api-sfn-function.ts [hunk 1]
```diff
@@ -28,6 +31,11 @@ export interface ApiSfnFunctionProps {
    * is called.
    */
   readonly stateMachine?: sfn.IStateMachine;
+
+  /**
+   * The AtatNetStack associated with this deployment (if present)
+   */
+  readonly network?: AtatNetStack;
```
[the remark this answers is not in the document]
Comment (jeffsegal-ccpo) [reply lib/constructs/api-sfn-function.ts]: Fixed

**Comment by jasonburkert-ccpo** [comment]
My understanding is that this will not be merged until the completion of AT-7326 in order to keep dev API accessible.

**Review by kylelaker-ccpo — APPROVED** [review]
Most of this has been done in other tickets at this point. I have cleaned this up a bit and simplified it to work in the updated code base. In all, the adjustments are pretty trivial.

**Pull request merged**